argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from collections import defaultdict
+from tempfile import TemporaryDirectory
+from typing import Any, Type, TYPE_CHECKING, Optional, Union
+from uuid import UUID
+
+from argilla.records import Record
+from argilla.responses import Response
+from argilla._exceptions import NotFoundError
+from argilla._helpers import resolve_hf_datasets_type
+
+if TYPE_CHECKING:
+    from argilla import Dataset
+    from argilla import Argilla
+    from argilla import Workspace
+
+HFDataset = resolve_hf_datasets_type()
+
+
+class HubImportExportMixin:
+    _default_dataset_repo_dir_name = ".argilla"
+    _default_configuration_files = ["settings.json", "dataset.json"]
+
+    def to_hub(
+        self: "Dataset",
+        repo_id: str,
+        generate_card: Optional[bool] = True,
+        *args,
+        **kwargs,
+    ) -> None:
+        """Pushes the `Dataset` to the Hugging Face Hub. If the dataset has been previously pushed to the
+        Hugging Face Hub, it will be updated instead of creating a new dataset repo.
+
+        Parameters:
+            repo_id: the ID of the Hugging Face Hub repo to push the `Dataset` to.
+            generate_card: whether to generate a dataset card for the `FeedbackDataset` in the Hugging Face Hub. Defaults
+                to `True`.
+            *args: the args to pass to `datasets.Dataset.push_to_hub`.
+            **kwargs: the kwargs to pass to `datasets.Dataset.push_to_hub`.
+        """
+
+        from huggingface_hub import HfApi, DatasetCardData
+
+        from argilla.datasets._export.card import (
+            ArgillaDatasetCard,
+            size_categories_parser,
+        )
+
+        hf_api = HfApi()
```

Comment:
better practice to set the `token` once and for all in `HfApi` definition

```suggestion
        hf_api = HfApi(token=kwargs.get("token"))
```